### PR TITLE
Improve public syntax information API 

### PR DESCRIPTION
**Pull request by NickAcPT**
### Description

This PR introduces some changes that will allow documentation extraction tools like ResourceSync(skUnity), [SkriptHubDocsTool](https://github.com/SkriptHub/SkriptHubDocsTool) (Skript Hub) and [DocExtractionTool](https://github.com/SkriptInsight/DocExtractionTool)(SkriptInsight) to have a more consistent API for extracting information. They, and new others will benefit from these changes.

Some of the changes include the following:
 - Store and expose Skript Expression types (The expression order is different depending on the expression's type)
 - Expose Syntax Element information with getters
 - Expose Event Value information with getters
 - Expose Language information with getters
 - Expose Function Parameter information with getters

### Why this change?

When tools need to fetch information from Skript, they have to resort to using Reflection.

https://github.com/SkriptHub/SkriptHubDocsTool/blob/70ae1be649a6797e98e2b108ebaee518d6789f65/src/main/java/net/skripthub/docstool/utils/EventValuesGetter.java#L28-L40

---
**Target Minecraft Versions:** any

### CODE REVIEW

**Review by ShaneBeee — APPROVED** [review]
Looks good to me. I would like to see what other members of the team have to say before merging.

**Review by ShaneBeee — APPROVED** [review]
The base branch was changed.

**Comment by NickAcPT** [comment]
@ShaneBeee 
> The base branch was changed.

Fixed it, but almost forgot to ask.. any particular reason as to why? ~~I would've asked on Discord, but you know the drill.~~

Either way, thanks for your review. If there's anything I missed in regards of exposing Syntax information, just let me know.

**Comment by ShaneBeee** [comment]
> Fixed it, but almost forgot to ask.. any particular reason as to why?

Are you referring to the base branch change? 
Master is going to be our release branch, where as the dev branches will be the branches we work on, until we are ready to release.

File: src/main/java/ch/njol/skript/localization/Language.java [hunk 1]
```diff
@@ -62,6 +62,32 @@
...
+		return new HashMap<>(english);
+	}
+	
+	/**
+	 * Get elements of the Localized language of Skript syntax.
+	 * @return A copy of all the elements of the Localized language
+	 */
+	@Nullable
+	public static HashMap<String, String> getLocalized() {
+		if (localized != null) return new HashMap<>(localized);
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/localization/Language.java:87]: return should be on its own line

File: src/main/java/ch/njol/skript/registrations/EventValues.java [hunk 2]
```diff
@@ -56,12 +58,73 @@ public EventValueInfo(Class<E> event, Class<T> c, Getter<T, E> getter, @Nullable
...
+		}
+		
+		/**
+		 * Get the classes that represent the excluded for this Event value.
+		 * @return The classes of the Excludes associated with this event value
+		 */
+		@Nullable
+		@SuppressWarnings("null")
+		public Class<? extends E>[] getExcludes() {
+			if (excludes != null) return Arrays.copyOf(excludes, excludes.length);
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/registrations/EventValues.java:85]: same as above

**Review by APickledWalrus — CHANGES_REQUESTED** [review]
two small things

File: src/main/java/ch/njol/skript/registrations/EventValues.java (hunk 2 above)
Comment (NickAcPT) [reply src/main/java/ch/njol/skript/registrations/EventValues.java]: Fixed this in abb07d1.

File: src/main/java/ch/njol/skript/localization/Language.java (hunk 1 above)
Comment (NickAcPT) [reply src/main/java/ch/njol/skript/localization/Language.java]: Fixed this in abb07d1.

**Review by APickledWalrus — APPROVED** [review]
LGTM 🙂

**Review by ShaneBeee — APPROVED** [review]

**Comment by ShaneBeee** [comment]
Im not sure If you turned off "Allow Edits" (the pic is an example from one of my PRs)
Screen Shot 2020-11-25 at 12 00 41 PM

But I don't have an option to update the branch:
Screen Shot 2020-11-25 at 11 59 07 AM

Could you please update your fork/branch to the dev-2.5 branch :)
Just want to make sure everything is A-OK before merging!

**Comment by NickAcPT** [comment]
> Im not sure If you turned off "Allow Edits" (the pic is an example from one of my PRs)
> Screen Shot 2020-11-25 at 12 00 41 PM
> 
> But I don't have an option to update the branch:
> Screen Shot 2020-11-25 at 11 59 07 AM
> 
> Could you please update your fork/branch to the dev-2.5 branch :)
> Just want to make sure everything is A-OK before merging!

That's odd.. I don't have that option on my side.. But I merged it anyway. I'm so sorry for that trouble and thanks for your time.

**Comment by ShaneBeee** [comment]
> > Im not sure If you turned off "Allow Edits" (the pic is an example from one of my PRs)
> > Screen Shot 2020-11-25 at 12 00 41 PM
> > But I don't have an option to update the branch:
> > Screen Shot 2020-11-25 at 11 59 07 AM
> > Could you please update your fork/branch to the dev-2.5 branch :)
> > Just want to make sure everything is A-OK before merging!
> 
> That's odd.. I don't have that option on my side.. But I merged it anyway. I'm so sorry for that trouble and thanks for your time.

No need to apologize. Thanks for the quick update 👍

**Pull request merged**